app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+                $validSchema = false;
+                $validData = false;
+            } elseif (! $this->validateData($properties, $fieldsToValidate)) {
+                $validData = false;
+            }
+            $insertionSchemaSuccess = $this->insertCriteriaSite($polygonUuid, $SCHEMA_CRITERIA_ID, $validSchema);
+            $insertionDataSuccess = $this->insertCriteriaSite($polygonUuid, $DATA_CRITERIA_ID, $validData);
+
+            $sitePolygon = new SitePolygon();
+            $sitePolygon->uuid = Str::uuid();
+            $sitePolygon->project_id = $properties['project_id'] ?? null;
+            $sitePolygon->proj_name = $properties['proj_name'] ?? null;
+            $sitePolygon->org_name = $properties['org_name'] ?? null;
+            $sitePolygon->country = $properties['country'] ?? null;
+            $sitePolygon->poly_id = $polygonUuid ?? null;
+            $sitePolygon->poly_name = $properties['poly_name'] ?? null;
+            $sitePolygon->site_id = $properties['site_id'] ?? null;
+            $sitePolygon->site_name = $properties['site_name'] ?? null;
+            $sitePolygon->poly_label = $properties['poly_label'] ?? null;
+            $sitePolygon->plantstart = ! empty($properties['plantstart']) ? $properties['plantstart'] : null;
+            $sitePolygon->plantend = ! empty($properties['plantend']) ? $properties['plantend'] : null;
+            $sitePolygon->practice = $properties['practice'] ?? null;
+            $sitePolygon->target_sys = $properties['target_sys'] ?? null;
+            $sitePolygon->distr = $properties['distr'] ?? null;
+            $sitePolygon->num_trees = $properties['num_trees'] ?? null;
+            $sitePolygon->est_area = $area ?? null;
+            $sitePolygon->created_at = now();
+            $sitePolygon->updated_at = now();
+            $sitePolygon->save();
+
+            return 'has saved correctly';
+        } catch (\Exception $e) {
+            return $e->getMessage();
+        }
+    }
+
+    public function getGeometryProperties(string $geojsonFilename)
+    {
+        $geojsonData = Storage::get("public/geojson_files/{$geojsonFilename}");
+        $geojson = json_decode($geojsonData, true);
+        if (! isset($geojson['features'])) {
+            return ['error' => 'GeoJSON file does not contain features'];
+        }
+
+        $propertiesList = [];
+        foreach ($geojson['features'] as $feature) {
+            $properties = $feature['properties'];
+            $geometryType = $feature['geometry']['type'];
+
+            if ($geometryType === 'Polygon' || $geometryType === 'MultiPolygon') {
+                $propertiesList[] = $properties;
+            }
+        }
+
+        return $propertiesList;
+    }
+
+    public function uploadKMLFile(Request $request)
+    {
+        if ($request->hasFile('file')) {
+            $kmlfile = $request->file('file');
+            $directory = storage_path('app/public/kml_files');
```

Comment:
Maybe it's lower in the PR, but I'm going to drop a question here in case not: Is anything being done to clean up these files? I wouldn't want us to run out of space on the machines because we weren't monitoring this or cleaning it up over time. If they need to be kept long term, they should be getting sent to S3, rather than the local disk (nothing we need to keep long term should be stored on the EC2 machines; they should be considered ephemeral).

Reply:
Right now we haven't got any work on that. But yes, files should be discarded after processing.
We need to create a new ticket for that.

Reply:
Ok, please link here when the ticket is created.

Reply:
Here is the [ticket](https://gfw.atlassian.net/browse/TM-886)